gpflux/types.py

```diff
@@ -19,9 +19,22 @@
 from typing import List, Tuple, Union
 
 import tensorflow as tf
+import tensorflow_probability as tfp
 
 from gpflow.base import TensorType
 
+
+def unwrap_dist(dist: tfp.distributions.Distribution) -> tfp.distributions.Distribution:
+    """
+    Unwrap the given distribution, if it is wrapped in a ``_TensorCoercible``.
+    """
+    while True:
```

Comment:
Are we sure this will always terminate? Would it be worth throwing an error after traversing 10 levels, for example?

Reply:
> Are we sure this will always terminate? Would it be worth throwing an error after traversing 10 levels, for example?

Dunno. Seems needlessly complicated to me. And are we sure there are no legitimate reasons for 10 levels of wrapping? Remember 640K ought to be enough for anybody.